src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -99,10 +106,127 @@ private static bool TryDecodeSuppressMessageAttributeData (CustomAttribute attri
 
 		private bool IsLocallySuppressed (int id, ICustomAttributeProvider provider, out SuppressMessageInfo info)
 		{
-			Dictionary<int, SuppressMessageInfo> suppressions;
 			info = default;
-			return _localSuppressions.TryGetValue (provider, out suppressions) &&
+			if (provider == null)
+				return false;
+
+			return _localSuppressions.TryGetValue (provider, out var suppressions) &&
 				suppressions.TryGetValue (id, out info);
 		}
+
+		private bool IsGloballySuppressed (int id, ICustomAttributeProvider provider, out SuppressMessageInfo info)
+		{
+			DecodeGlobalSuppressMessageAttributes (provider);
+			return _globalSuppressions.HasGlobalSuppressions (id, provider, out info);
+		}
+
+		private void DecodeGlobalSuppressMessageAttributes (ICustomAttributeProvider provider)
+		{
+			ModuleDefinition module;
+			switch (provider.MetadataToken.TokenType) {
+			case TokenType.Module:
+				module = provider as ModuleDefinition;
+				break;
+			case TokenType.Assembly:
+				module = (provider as AssemblyDefinition).MainModule;
+				break;
+			case TokenType.TypeDef:
+				module = (provider as TypeDefinition).Module;
+				break;
+			case TokenType.Method:
+			case TokenType.Property:
+			case TokenType.Field:
+			case TokenType.Event:
+				module = (provider as IMemberDefinition).DeclaringType.Module;
+				break;
+			default:
+				_context.LogMessage (MessageContainer.CreateInfoMessage ("`UnconditionalSuppressMessage` attribute was placed in an language element which is currently not supported."));
+				return;
+			}
+
+			Debug.Assert (module != null);
+			AssemblyDefinition assembly = module.Assembly;
+			if (!_globalSuppressions.InitializedAssemblies.Contains (assembly)) {
+				DecodeGlobalSuppressMessageAttributes (assembly);
+				_globalSuppressions.InitializedAssemblies.Add (assembly);
+			}
+		}
+
+		private void DecodeGlobalSuppressMessageAttributes (AssemblyDefinition assembly)
+		{
+			void LookForSuppressions (ICustomAttributeProvider provider) {
```

Comment:
Extract this to a static method outside - I don't see why this needs to be a local method.